core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -385,9 +390,33 @@ private[spark] class MemoryStore(
     }
   }
 
+  def manual_close[T <: MemoryEntry[_]](entry: T): T = {
+    val entryManualCloseTasks = Future {
+      entry match {
+        case e: DeserializedMemoryEntry[_] => e.value.foreach {
+          case o: AutoCloseable =>
+            try {
+              o.close
+            } catch {
+              case NonFatal(e) =>
+                logWarning(s"Got NonFatal exception during remove")
+            }
+          case _ =>
+        }
+        case _ =>
+      }
+    }
+    entryManualCloseTasks.onComplete {
+      case Success(_) =>
+      case Failure(e) => throw e
+    }
```

Comment:
@srowen , I replace ThreadUtils.awaitResult with onComplete here. And I also wrapped manual close as a function upon @mridulm 's suggestion.